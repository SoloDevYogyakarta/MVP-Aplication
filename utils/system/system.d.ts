declare function joinpath(path: string): string;
declare function readpath(path: string): string;
declare function createpath(path: string, data: unknown): void;
declare function removepath(path: string): void;
export { joinpath, readpath, createpath, removepath };
