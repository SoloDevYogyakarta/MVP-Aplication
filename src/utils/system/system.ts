import fs from 'fs';
import { join } from 'path';

function joinpath(path: string): string {
  return join(__dirname, path);
}

function readpath(path: string): string {
  return fs.readFileSync(joinpath(path), { encoding: 'utf-8' });
}

function createpath(path: string, data: unknown): void {
  let values = data;
  if (typeof values !== 'string') {
    values = JSON.stringify(values);
  }
  fs.writeFileSync(joinpath(path), values as string);
}

function removepath(path: string): void {
  fs.unlinkSync(joinpath(path));
}

export { joinpath, readpath, createpath, removepath };
