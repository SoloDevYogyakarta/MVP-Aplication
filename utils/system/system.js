"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.removepath = exports.readpath = exports.createpath = exports.joinpath = void 0;
const fs_1 = __importDefault(require("fs"));
const path_1 = require("path");
function joinpath(path) {
    return (0, path_1.join)(__dirname, path);
}
exports.joinpath = joinpath;
function createpath(path, data) {
    if (typeof data !== 'string') {
        data = JSON.stringify(data);
    }
    fs_1.default.writeFileSync(joinpath(path), data);
}
exports.createpath = createpath;
function readpath(path) {
    return fs_1.default.readFileSync(joinpath(path), { encoding: 'utf-8' });
}
exports.readpath = readpath;
function removepath(path) {
    fs_1.default.unlinkSync(joinpath(path));
}
exports.removepath = removepath;
//# sourceMappingURL=system.js.map