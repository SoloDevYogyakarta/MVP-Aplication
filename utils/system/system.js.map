{"version":3,"file":"system.js","sourceRoot":"","sources":["../../../src/utils/system/system.ts"],"names":[],"mappings":";;;;;;AAAA,4CAAoB;AACpB,+BAA4B;AAE5B,SAAgB,QAAQ,CAAC,IAAY;IACnC,OAAO,IAAA,WAAI,EAAC,SAAS,EAAE,IAAI,CAAC,CAAC;AAC/B,CAAC;AAFD,4BAEC;AAED,SAAgB,UAAU,CAAC,IAAY,EAAE,IAAa;IACpD,IAAI,OAAO,IAAI,KAAK,QAAQ,EAAE;QAC5B,IAAI,GAAG,IAAI,CAAC,SAAS,CAAC,IAAI,CAAC,CAAC;KAC7B;IACD,YAAE,CAAC,aAAa,CAAC,QAAQ,CAAC,IAAI,CAAC,EAAE,IAAc,CAAC,CAAC;AACnD,CAAC;AALD,gCAKC;AAED,SAAgB,QAAQ,CAAC,IAAY;IACnC,OAAO,YAAE,CAAC,YAAY,CAAC,QAAQ,CAAC,IAAI,CAAC,EAAE,EAAE,QAAQ,EAAE,OAAO,EAAE,CAAC,CAAC;AAChE,CAAC;AAFD,4BAEC;AAED,SAAgB,UAAU,CAAC,IAAY;IACrC,YAAE,CAAC,UAAU,CAAC,QAAQ,CAAC,IAAI,CAAC,CAAC,CAAC;AAChC,CAAC;AAFD,gCAEC"}