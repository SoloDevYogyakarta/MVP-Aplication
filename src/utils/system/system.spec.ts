import * as system from './system';

describe('System', () => {
  it('should to be defined', () => expect(system).toBeDefined());

  it('render correctly', () => expect(system).toMatchSnapshot());

  it('createpath should to be successfully', () => {
    const create = system.createpath('../folder-text/object.txt', {
      message: 'works!',
    });
    system.createpath('../folder-text/remove.txt', 'remove');
    system.createpath('../folder-text/fake.txt', 'fake');
    expect(create).toEqual(undefined);
  });

  it('joinpath should to be successfully', () =>
    expect(system.joinpath('../folder-text')).toContain('folder-text'));

  it('readpath should to be successfully', () =>
    expect(JSON.parse(system.readpath('../folder-text/object.txt'))).toEqual({
      message: 'works!',
    }));

  it('removepath should to be successfully', () =>
    expect(system.removepath('../folder-text/remove.txt')).toEqual(undefined));
});
