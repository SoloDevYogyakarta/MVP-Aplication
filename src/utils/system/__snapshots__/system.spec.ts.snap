// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`System render correctly 1`] = `
{
  "createpath": [Function],
  "joinpath": [Function],
  "readpath": [Function],
  "removepath": [Function],
}
`;
